waku/handshake.go

```diff
@@ -18,14 +18,61 @@ import (
 // In the case of RLP, options should be serialized to an array of tuples
 // where the first item is a field name and the second is a RLP-serialized value.
 type statusOptions struct {
-	PoWRequirement       uint64      `rlp:"key=0"` // RLP does not support float64 natively
+	PoWRequirement       *uint64     `rlp:"key=0"` // RLP does not support float64 natively
 	BloomFilter          []byte      `rlp:"key=1"`
-	LightNodeEnabled     bool        `rlp:"key=2"`
-	ConfirmationsEnabled bool        `rlp:"key=3"`
-	RateLimits           RateLimits  `rlp:"key=4"`
+	LightNodeEnabled     *bool       `rlp:"key=2"`
+	ConfirmationsEnabled *bool       `rlp:"key=3"`
+	RateLimits           *RateLimits `rlp:"key=4"`
 	TopicInterest        []TopicType `rlp:"key=5"`
 }
 
+func (s statusOptions) WithoutDefaults() statusOptions {
+	if s.PoWRequirement != nil && *s.PoWRequirement == 0 {
+		s.PoWRequirement = nil
+	}
+
+	if s.LightNodeEnabled != nil && !*s.LightNodeEnabled {
+		s.LightNodeEnabled = nil
+	}
+
+	if s.ConfirmationsEnabled != nil && !*s.ConfirmationsEnabled {
+		s.ConfirmationsEnabled = nil
+	}
+
+	if s.RateLimits != nil && s.RateLimits.IsZero() {
+		s.RateLimits = nil
+	}
+
+	return s
+}
+
+func (s statusOptions) WithDefaults() statusOptions {
+	if s.PoWRequirement == nil {
+		var pow uint64 = 0
+		s.PoWRequirement = &pow
+	}
+
+	if s.LightNodeEnabled == nil {
+		lightNodeEnabled := false
+		s.LightNodeEnabled = &lightNodeEnabled
+	}
+
+	if s.ConfirmationsEnabled == nil {
+		confirmationsEnabled := false
+		s.ConfirmationsEnabled = &confirmationsEnabled
+	}
+
+	if s.RateLimits == nil {
+		s.RateLimits = &RateLimits{}
```

Comment:
What does this default result in?

Reply:
I believe it means no rate limits, but @adambabik can confirm

Reply:
Correct. I will start working on full rate limiting support and the idea is to verify it using `RateLimits.IsZero() == false`.